src/mpid/ch4/src/ch4_types.h

```diff
@@ -91,6 +91,8 @@ enum {
 
     MPIDIG_COMM_ABORT,
 
+    MPIDI_OFI_INTERNAL_HANDLER_CONTROL,
```

Comment:
Will this OFI handler disappear from here eventually?

Reply:
It could disappear by using the dynamic registration.

Personally, I don't really mind having static IDs in a global table. The benefits include:
* We easily can browse for all existing active message protocols, helps us to spot similarity and avoid unnecessary duplication.
* Static table makes debug easier. We can check the handler_id and map back to the protocol.
* Much simpler and clearer mechanism.

But if someone insist on hiding device symbols, then we can hide it in a separate commit.

Reply:
My concern is having netmod specific handlers exposed to the generic layer. The reason I asked is because UCX does not define additional handlers and uses the generic ones. Is the idea here to eventually have OFI behave similarly to UCX?

Reply:
> Is the idea here to eventually have OFI behave similarly to UCX?

Not really. 

If anything, I am aiming to put OFI/UCX/POSIX internal am protocols at the same level of generic protocols. Whether we want to segregate the symbols is just a trivial matter. Each protocol will be completely isolated anyway, so they can be placed anywhere.

Reply:
Which means that there will not be any netmod specific protocol in the generic layer

Reply:
```
    ---------------------        -----------------
    |  generic protocols  |       |  OFI protocols   |
    ---------------------        -----------------
    -------------------------------------------------
   |    OFI transport (may call generic/OFI protocols |
   --------------------------------------------------
```
Does this make sense?